### ISSUE TITLE
Can not create CampaignBudget on version 5.1.1

### ISSUE THREAD

**Issue by nmluancs** [Opteo/google-ads-api#243]
After upgrading google-ads-api to 5.1.1, i can not create a campaign budget:
If i did not set total_amount_micros, i got error total_amount_micros: Too low
If it set total_amount_micros, i got error: 

```
GoogleAdsError {
  error_code: ErrorCode { request_error: 1 },
  message: 'The error code is not in this version.',
  trigger: Value { string_value: '' },
  location: ErrorLocation {
    field_path_elements: [ 
       FieldPathElement { field_name: 'operations', index: 0 },
       FieldPathElement { field_name: 'create' },
       FieldPathElement { field_name: 'total_amount_micros' }
    ]
  }
}
```
Here is my testing code:
```ts
const campaignBudget = new resources.CampaignBudget({
      name: 'TestA1',
      amount_micros: toMicros(500000),
      delivery_method: "STANDARD",
      explicitly_shared: false,
      period: "DAILY",
      // total_amount_micros: toMicros(500000 * 30.4),
      type: "STANDARD",
    });
    const options: MutateOptions = {
      validate_only: true
    };

    try {
      const res = await adsCustomer.Customer.campaignBudgets.create([campaignBudget], options);
    } catch(ex) {}
```

**Comment by nmluancs** [comment]
It seems there is problems in converting between resource interface to entity class.
If i use resources.ICampaignBudget, it works.

const campaignBudget : new resources.ICampaignBudget = {
      name: 'TestA1',
      amount_micros: toMicros(500000),
      delivery_method: "STANDARD",
      explicitly_shared: false,
      period: "DAILY",
      // total_amount_micros: toMicros(500000 * 30.4),
      type: "STANDARD",
    };
    const options: MutateOptions = {
      validate_only: true
    };

    try {
      const res = await adsCustomer.Customer.campaignBudgets.create([campaignBudget], options);
    } catch(ex) {}

**Comment by nmluancs** [comment]
Hello devs,
After reading source code, i think the problem is that in interface, all fields allow null but when implement in class those fields no longer support null.
For example: 
- in CampaignBudget : amount_micros, total_amount_micros has type of number with default value is 0 so that if i did not set total_amount_micros, i will get the error (total_amount_micros too low)

This problem happens in all implemented class and it takes a lot of bugs.
Please have a look. Thank you

**Comment by nmluancs** [comment]
Hmm, sorry. It seems Google protobuf doesn't support nullable field

**Comment by WillCooter** [comment]
Hi @nmluancs. We are aware of this issue and are waiting on a fix in the [node-js gapic library](https://github.com/googleapis/gapic-generator-typescript). Once this has been done we will release with the fix.

**Comment by rosslavery** [comment]
Just to chime in that I'm affected by this as well. Also getting other errors for atomic mutations that weren't there before our migration. We've never had to specify fields like `final_url_suffix` while creating a campaign before.

Not sure if this is related to the gapic library. If not I can open a separate issue. In the meantime we've rolled back to library 4.x.

E.g.:

```
{
      "error_code": {
        "field_error": "INVALID_VALUE"
      },
      "message": "The field's value is invalid.",
      "trigger": {
        "string_value": ""
      },
      "location": {
        "field_path_elements": [
          {
            "field_name": "mutate_operations",
            "index": 2
          },
          {
            "field_name": "campaign_operation"
          },
          {
            "field_name": "create"
          },
          {
            "field_name": "final_url_suffix"
          }
        ]
      }
    },
```

**Comment by Zikoel** [comment]
Good news, the [googleapis/gapic-generator-typescript#848](https://github.com/googleapis/gapic-generator-typescript/issues/848) was marked as done!!

**Comment by WillCooter** [comment]
Yes indeed. We will implement that along with [google-ads v7](https://developers.google.com/google-ads/api/docs/release-notes) next week.

**Comment by WillCooter** [comment]
@nmluancs @rosslavery @Zikoel The latest version of the API has been released under `v5.2.0` of this library. Upgrading should hopefully fix your issue. If not then feel free to reopen this issue.

**Comment by spicy-sauce** [comment]
Hello, 
I'm using version 5.2.0 and when trying to set a campaignBudget with total_amount_micros, getting this result:
```
  errors: [
    GoogleAdsError {
      error_code: ErrorCode { request_error: 1 },
      message: 'The error code is not in this version.',
      trigger: Value { string_value: '' },
      location: ErrorLocation {
        field_path_elements: [
          FieldPathElement { field_name: 'operations', index: 0 },
          FieldPathElement { field_name: 'create' },
          FieldPathElement { field_name: 'total_amount_micros' }
        ]
      }
    }
  ],
```
